### DISCUSSION TITLE
Is there an example how to define a file upload

### DISCUSSION THREAD

**Discussion by mmogylenko** [swaggest/openapi-go#131] [Q&A]
```go
type upload struct { File `formData:"file" required:"true" description:"The content of the file to upload."` }


op, _ := reflector.NewOperationContext(http.MethodPost, "/api/upload")

op.AddReqStructure(
     new(reportUploadRequest),
     openapi.WithContentType("multipart/form-data"),
)
```

Will produce this:
![Screenshot 2024-09-30 at 6 58 16 PM](https://github.com/user-attachments/assets/c35e541e-66a6-40e8-81a4-f06ada038905)
and not upload form like this:
![Screenshot 2024-09-30 at 6 58 01 PM](https://github.com/user-attachments/assets/be1cb083-b20f-4d5c-8f34-7a751b7a9fc5)

**Comment by vearutop** [comment]
Hello, file uploads are inferred from `*multipart.FileHeader` or `multipart.File`.
Please check an example in `swaggest/rest`:
https://github.com/swaggest/rest/blob/v0.2.66/_examples/advanced-generic-openapi31/file_upload.go#L17-L18
```go
	type upload struct {
		Simple  string                `formData:"simple" description:"Simple scalar value in body."`
		Query   int                   `query:"in_query" description:"Simple scalar value in query."`
		Upload1 *multipart.FileHeader `formData:"upload1" description:"Upload with *multipart.FileHeader."`
		Upload2 multipart.File        `formData:"upload2" description:"Upload with multipart.File."`
	}
```

and also https://github.com/swaggest/rest/blob/v0.2.66/_examples/advanced-generic-openapi31/file_multi_upload.go#L17-L18.
